### PR TITLE
Wifi: Change Lottery winner logic

### PR DESCRIPTION
**Pull request by Rixius**
After a Policy decision within Wi-Fi Staff, it's been decided to update the
Lottery Giveaway to still pick winners when there are less than maxWinners,
but people entered. This will allow low entry GA's to overestimate and still
have winners.

### CODE REVIEW

File: server/chat-plugins/wifi.ts [hunk 1]
```diff
@@ -437,13 +437,13 @@ export class LotteryGiveaway extends Giveaway {
...
-		if (userlist.length < this.maxWinners) {
+		if (userlist.length === 0) {
 			this.changeUhtml('<p style="text-align:center;font-size:13pt;font-weight:bold;">The giveaway was forcibly ended.</p>');
 			delete this.room.giveaway;
-			return this.room.send("The giveaway has been forcibly ended as there are not enough participants.");
+			return this.room.send("The giveaway has been forcibly ended as there are no participants.");
 		}
 
-		while (this.winners.length < this.maxWinners) {
+		while (this.winners.length < this.maxWinners && this.userlist > 0) {
```
Comment (AnnikaCodes) [inline server/chat-plugins/wifi.ts:446]:
```suggestion
		while (this.winners.length < this.maxWinners && userlist.length > 0) {
```

Comment (AnnikaCodes) [reply server/chat-plugins/wifi.ts]: Never mind, looks like you figured it out :)

Comment (Rixius) [reply server/chat-plugins/wifi.ts]: Yes, I hadn't set the project up locally to run the tests myself.
Seeing the logical issue, I set the repository up locally, fixed this issue,
and have tested, rebased and re-submitted.

Comment (AnnikaCodes) [reply server/chat-plugins/wifi.ts]: Testing locally is always a good plan.

Comment (AnnikaCodes) [reply server/chat-plugins/wifi.ts]: I thought I already told you this, but I ran the checks locally and they're passing.

**Comment by Zarel** [comment]
So what's the status on this?

**Comment by Rixius** [comment]
After the force-push from 394000f to d752701 on July 6th, it was passing tests locally on my machine, and AnnikaCodes', but appeared to cause some sort of issue in Github's CI machines. Annika said on Showdown #development that there wasn't anything more for me to handle.

Currently pulling the latest master and running tests to confirm that tests are still passing, and then I will re-push and see if that causes Github's testing to run the right code.

**Comment by Zarel** [comment]
@AnnikaCodes does this look good to you?

**Pull request merged**